### PR TITLE
SNOW-499493 Add Interval Precision to Timestamp

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-499493

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/connector/arrow_context.py [hunk 1]
```diff
@@ -88,12 +88,30 @@ def TIMESTAMP_NTZ_to_python(self, microseconds):
     def TIMESTAMP_NTZ_to_python_windows(self, microseconds):
         return ZERO_EPOCH + timedelta(seconds=microseconds)
 
-    def TIMESTAMP_LTZ_to_python(self, microseconds):
+    def TIMESTAMP_LTZ_to_python_one_field(self, microseconds):
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/arrow_context.py:91]: Missing return type-hint and confusing name

File: src/snowflake/connector/arrow_context.py [hunk 2]
```diff
@@ -88,12 +88,30 @@ def TIMESTAMP_NTZ_to_python(self, microseconds):
...
         tzinfo = self._get_session_tz()
         return datetime.fromtimestamp(microseconds, tz=tzinfo)
 
-    def TIMESTAMP_LTZ_to_python_windows(self, microseconds):
+    def TIMESTAMP_LTZ_to_python_two_field(self, epoch: float, fraction: float):
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/arrow_context.py:95]: Please add a return type-hint and change the name of this function to something that describes how this is different from the other function

File: src/snowflake/connector/arrow_context.py [hunk 3]
```diff
@@ -88,12 +88,30 @@ def TIMESTAMP_NTZ_to_python(self, microseconds):
...
         tzinfo = self._get_session_tz()
+        return datetime.fromtimestamp(epoch, tz=tzinfo) + timedelta(seconds=fraction)
+
+    def TIMESTAMP_LTZ_to_python_one_field_windows(self, microseconds):
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/arrow_context.py:99]: Add type-hints and maybe a doc-string that describes why this needs to act differently on Windows

File: test/integ/test_arrow_result.py [hunk 4]
```diff
@@ -464,7 +464,9 @@ def test_select_timestamp_with_scale(conn_cnx, scale, type):
     sql_text = "select a from {} order by s".format(table)
     row_count = len(cases) + 2
     col_count = 1
-    iterate_over_test_chunk(type, conn_cnx, sql_text, row_count, col_count)
+    iterate_over_test_chunk(
+        type, conn_cnx, sql_text, row_count, col_count, eps=timedelta(microseconds=1)
```
Comment (sfc-gh-mkeller) [inline test/integ/test_arrow_result.py:468]: As we talked about this, add a Jira in the code here that describes fixing the JSON implementation too

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]

**Comment by sfc-gh-mkeller** [comment]
recheck

**Comment by sfc-gh-jbahk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**